plugins/woocommerce-blocks/assets/js/blocks/product-gallery/frontend.tsx

```diff
@@ -205,6 +217,47 @@ const productGallery = {
 				'wc-block-product-gallery-dialog-open'
 			);
 		},
+		onTouchStart: ( event: TouchEvent ) => {
+			const context = getContext();
+			context.touchStartX = event.touches[ 0 ].clientX;
+			context.touchCurrentX = event.touches[ 0 ].clientX;
+			context.isDragging = true;
+		},
+		onTouchMove: ( event: TouchEvent ) => {
+			const context = getContext();
+			if ( ! context.isDragging ) {
+				return;
+			}
+			context.touchCurrentX = event.touches[ 0 ].clientX;
+			event.preventDefault();
+		},
+		onTouchEnd: () => {
+			const context = getContext();
+			if ( ! context.isDragging ) {
+				return;
+			}
+
+			const SNAP_THRESHOLD = 0.3;
```

Comment:
After playing with it, I'd definitely decrease the threshold, 0.2 sounds good. It may annoy users when every other swipe doesn't change the image. (I was that user 😄 )

Reply:
https://github.com/woocommerce/woocommerce/pull/55010/commits/c2141183621d65303f08c5af81be8531e36317ea